pkg/apis/hive/v1/clusterdeployment_types.go

```diff
@@ -122,6 +112,14 @@ type Provisioning struct {
 	// Any version of InstallConfig can be used, provided it can be parsed by the openshift-install
 	// version for the release you are provisioning.
 	InstallConfigSecret corev1.LocalObjectReference `json:"installConfigSecret"`
+
+	// SSHPrivateKeySecret is the reference to the secret that contains the private SSH key to use
+	// for access to compute instances. This private key should correspond to the public key included
+	// in the InstallConfig. The private key is used by Hive to gather logs on the target cluster if
+	// there are install failures.
+	// The SSH private key is expected to be in the secret data under the "ssh-privatekey" key.
+	// +optional
+	SSHPrivateKeySecret *corev1.LocalObjectReference `json:"sshPrivateSecretKey,omitempty"`
```

Comment:
Got a little nervous about this change after hearing some of what Eric said in the meetings yesterday. (that we don't need to deal in private keys) I think our plan is ok though, if you don't want to give us a private key, give the InstallConfig the public key, and us nothing?

Or should be backtrack here and keep it as a ssh-key that might contain a public and/or private key key.